python/setuputils.py

```diff
@@ -14,30 +14,56 @@
 # limitations under the License.
 #
 
+import glob
 import os
 import re
+import shutil
 import subprocess
 import warnings
 
 
+def clean_folder(path):
+    """
+    Function to clean all Cython and Python artifacts and cache folders. It
+    clean the folder as well as its direct children (NOT recursively).
+
+    Parameters
+    ----------
+    path : String
+        Path to the folder to be cleaned.
+    """
+    shutil.rmtree(path + '/__pycache__', ignore_errors=True)
+
+    folders = glob.glob(path + '/*')
+    for folder in folders:
+        shutil.rmtree(folder + '/__pycache__', ignore_errors=True)
+
+        clean_folder(folder)
```

Comment:
This looks recursive to me? (contrary to the docstring) It calls clean_folder on each subfolder right?